index.js

```diff
@@ -53,42 +53,37 @@ function watch(glob, options, cb) {
   var queued = false;
   var running = false;
 
-  // These use sparse arrays to keep track of the index in the
-  // original globs array
-  var positives = new Array(glob.length);
-  var negatives = new Array(glob.length);
+  var positives = [];
+  var negatives = [];
 
   // Reverse the glob here so we don't end up with a positive
   // and negative glob in position 0 after a reverse
   glob.reverse().forEach(sortGlobs);
 
   function sortGlobs(globString, index) {
     var result = isNegatedGlob(globString);
-    if (result.negated) {
-      negatives[index] = result.pattern;
-    } else {
-      positives[index] = result.pattern;
-    }
+    var obj = { pattern: result.pattern, index: index };
+    (result.negated ? negatives : positives).push(obj);
   }
 
+  var toWatch = positives.map(getPattern);
+
   function shouldBeIgnored(path) {
-    var positiveMatch = anymatch(positives, path, true);
-    var negativeMatch = anymatch(negatives, path, true);
+    var negativeMatch = anymatch(negatives.map(getPattern), path, true);
```

Comment:
the result of negatives.map could be preprocessed and kept in the closure here for more speed, at the cost of more RAM usage. Should I do it?